File: ia/Configs.js
//
// Configs.js for super_zappy in /home/aracthor/programs/projects/hub/super_zappy/ia/scripts
// 
// Made by 
// Login   <aracthor@epitech.net>
// 
// Started on  Fri Oct 31 13:29:59 2014 
// Last Update Tue Nov  4 14:07:32 2014 
//

DEFAULT_HOST	= "127.0.0.1";
DEFAULT_PORT	= 4242;

function	Configs(argc, argv)
{
    this.host = DEFAULT_HOST;
    this.port = DEFAULT_PORT;
    this.name = null;

    this.reader = null;

    for (i = 1; i < argc; ++i)
    {
	this.readArg(argv[i]);
    }

    if (this.name == null)
    {
	throw new SyntaxError("Please give a team name. (with -f)");
    }
}


Configs.prototype.changePort = function(port)
{
    this.port = Number(port);
}

Configs.prototype.changeHost = function(host)
{
    this.host = host;
}

Configs.prototype.changeTeamName = function(name)
{
    if (this.name != null)
    {
	throw new SyntaxError("Please give only one team name.");
    }
    this.name = name;
}


Configs.prototype.readArg = function(arg)
{
    if (arg[0] == '-')
    {
	this.changeReader(arg[1]);
    }
    else if (this.reader == null)
    {
	throw new SyntaxError("Syntax error : missing flag before arg");
    }
    else
    {
	this.reader(arg);
    }
}

Configs.prototype.changeReader = function(flag)
{
    if (flag == 'p')
    {
	this.reader = this.changePort;
    }
    else if (flag == 'h')
    {
	this.reader = this.changeHost;
    }
    else if (flag == 'f')
    {
	this.reader = this.changeTeamName;
    }
    else
    {
	throw new SyntaxError("Syntax error : unknow flag " + flag);
    }
}

File: AI/python/Configs.py
#!/usr/bin/python3
## Configs.py for super_zappy in /home/aracthor/programs/projects/hub/super_zappy/ia/python
## 
## Made by aracthor
## Login   <aracthor@epitech.net>
## 
## Started on  Tue Feb 24 09:37:58 2015 aracthor
## Last Update Tue Feb 24 12:24:33 2015 aracthor
##

from    Exceptions      import SyntaxException

class   Configs:
    DEFAULT_HOST = "localhost"
    DEFAULT_PORT = 4242

    def __init__(self, argc, argv):
        self.host = self.DEFAULT_HOST
        self.port = self.DEFAULT_PORT
        self.script = None
        self.name = None

        self.reader = None

        self.readers = {
            'p': self.changePort,
            'h': self.changeHost,
            'n': self.changeName,
            'f': self.changeScript
        }

        for arg in argv[1:]:
            self.readArg(arg)

        if self.script == None:
            raise SyntaxException("Please give a script (with -f)")
        if self.name == None:
            raise SyntaxException("Please give a name (with -n)")


    def readArg(self, arg):
        if (arg[0] == '-'):
            self.changeReader(arg[1])
        elif (self.reader == None):
            raise SyntaxException("missing flag before arg")
        else:
            self.reader(arg)

    def changeReader(self, flag):
        if flag not in self.readers:
            raise SyntaxException("unknow flag " + flag)
        self.reader = self.readers[flag]

    
    def changePort(self, port):
        self.port = port
    
    def changeHost(self, host):
        self.host = host
    
    def changeName(self, name):
        self.name = name
    
    def changeScript(self, script):
        self.script = script
